src/DotNetty.Buffers/ByteBufferUtil.cs

```diff
@@ -569,5 +569,12 @@ static void AppendHexDumpRowPrefix(StringBuilder dump, int row, int rowStartInde
                 dump.Append('|');
             }
         }
+
+        public static string DecodeString(IByteBuffer src, int readerIndex, int len, Encoding encoding)
+        {
+            if (len == 0) return string.Empty;
+            
+            return encoding.GetString(src.Array.Slice(len));
```

Comment:
In netty there's a lot of plumbing around direct vs heap buffers which does not quite (yet) applies in DotNetty.
Check this line: https://github.com/netty/netty/blob/4.1/buffer/src/main/java/io/netty/buffer/ByteBufUtil.java#L541
This is quite close in semantics to `HasArray` in DotNetty. It's one of those methods where it's impossible to follow netty 1:1 just because underlying framework is different :)